click_odoo_contrib/initdb.py

```diff
@@ -0,0 +1,326 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# Copyright 2018 ACSONE SA/NV (<http://acsone.eu>)
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html).
+import contextlib
+from datetime import datetime, timedelta
+from fnmatch import fnmatch
+import hashlib
+import logging
+import os
+import re
+
+import click
+import click_odoo
+
+from click_odoo import odoo
+
+from .manifest import expand_dependencies
+
+_logger = logging.getLogger(__name__)
+
+
+EXCLUDE_PATTERNS = ('*.pyc', '*.pyo')
+
+
+def check_dbname(dbname):
+    if not re.match('^[A-Za-z][A-Za-z0-9-]*$', dbname):
```

Comment:
@sbidoul Is there a reason for character limitation ? e.g.: underscore is accepted in db name by postgres.

Reply:
underscore is also a wildcard character in sql like, that's why I avoided it